### ISSUE TITLE
`yarn -v` is equivalent to `yarn`

### ISSUE THREAD

**Issue by ithinkihaveacat** [yarnpkg/yarn#742]
**Do you want to request a *feature* or report a *bug*?**

Bug.

**What is the current behavior?**

Unknown arguments seem to be ignored, which leads to unexpected behavior like `yarn -v` installing modules.

**If the current behavior is a bug, please provide the steps to reproduce.**

Type `yarn -v` in a directory with a `package.json` file.

**What is the expected behavior?**

Either the version number is output (i.e. equivalent to `yarn -V` or `yarn --version`) or `yarn` warns about an unknown argument, and does nothing.

**Please mention your node.js, yarn and operating system version.**

```sh
$ node -v
v6.7.0
$ yarn -V
0.15.1
$ uname -a
Darwin kapow.local 16.0.0 Darwin Kernel Version 16.0.0: Mon Aug 29 17:56:20 PDT 2016; root:xnu-3789.1.32~3/RELEASE_X86_64 x86_64
```

**Comment by serkanyersen** [comment]
I don't know what is the exact reason but they [specifically allowed this](https://github.com/yarnpkg/yarn/blob/126d8c21f56466826e4cc9d43b66123872b982fe/src/cli/index.js#L60)

**Comment by s992** [comment]
Probably caused by [this bit](https://github.com/yarnpkg/yarn/blob/126d8c21f56466826e4cc9d43b66123872b982fe/src/cli/index.js#L92).

**Comment by mattapperson** [comment]
Having this issue too. Prevents me from having my yarn version displayed in my bash prompt as I do with my npm version now... 😞

**Comment by rjmunro** [comment]
@serkanyersen The reason is in the commit (use the blame button to find it): https://github.com/yarnpkg/yarn/commit/54a7900cd6d0c42051c62176f99b1ebf76c49779

> By not alllowing unknown options we break forwards and backwards compatibility.
> This settings allows options that are not supported yet.

I personally don't agree with that philosophy. I think it's much better that things fail loudly and you can then fix them, rather than things appear to work, but it turns out the didn't do what you expected because you are on the wrong version.

**Comment by ithinkihaveacat** [comment]
@rjmunro Interesting. Unfortunately that combined with a few other decisions results in `yarn -v` surprisingly doing work, when I was expecting it to either emit version information, or output help (but otherwise be a fast no-op).

I think it would be better if either:

1. `yarn -v` was changed to emit version information. (The same as `yarn -V` and `yarn --version`.)
1. Unknown options are not ignored, and instead emit errors.
1. The default command is `help`, not `install`.

It seems like (2) and (3) are both working as intended, so maybe the only fix is (1)?

**Comment by kittens** [comment]
Fixed via #777.
